pkg/apis/core/types.go

```diff
@@ -2953,6 +2953,25 @@ func (ps *PodSpec) Workloads() []CommonInfo {
 	return ps.WorkloadInfo
 }
 
+func (ps *PodSpec) SetWorkloads() {
+	// Update mutable fields
+	if len(ps.WorkloadInfo) != 0 {
```

Comment:
we have a commonInfo which can be used to avoid this if-else block, any reason it is not used here ?

Reply:
this is common info field.